design/how-ironic-works.md

```diff
@@ -146,6 +146,34 @@ Between ironic and ironic-inspector:
 * Each service must be able to reach the API endpoint for
   the the other service.
 
+## What is ironic-inspector?
+Ironic-inspector is an auxiliary service that provides separated API to
+inspect and register hardware properties for a bare metal node.
+This process may be triggered during discovery and allow automatic node
+registration in the ironic DB.
+The process of "hardware introspection" allows also to get the hardware
+parameters from a bare metal node and ready it for scheduling.
+Power management credentials should be already set to allow node
+introspection.
+
+Ironic-inspector relies on plugins for data processing even for standard
+functionality; basic hardware introspection plugins are enabled by
+default.
+
+Ironic-inspector supports the creation of custom introspection rules to
+run during the introspection process.
+
+### What are introspection rules?
```

Comment:
Are we planning to use introspection rules? If no, let's probably skip them?

Reply:
I will leave this out for now, we can always add this later if we need it